app/move/controllers/unassign.test.js

```diff
@@ -0,0 +1,277 @@
+        expect(controller.use.callCount).to.equal(1)
+      })
+    })
+
+    describe('#middlewareLocals()', function() {
+      beforeEach(function() {
+        sinon.stub(FormWizardController.prototype, 'middlewareLocals')
+        sinon.stub(controller, 'use')
+
+        controller.middlewareLocals()
+      })
+
+      it('should call parent method', function() {
+        expect(FormWizardController.prototype.middlewareLocals).to.have.been
+          .calledOnce
+      })
+
+      it('should call setMoveRelationships middleware', function() {
+        expect(controller.use.firstCall).to.have.been.calledWith(
+          controller.setMoveRelationships
+        )
+      })
+
+      it('should call correct number of middleware', function() {
+        expect(controller.use.callCount).to.equal(1)
+      })
+    })
+
+    describe('#checkAllocation()', function() {
+      let res, next
+
+      beforeEach(function() {
+        next = sinon.stub()
+      })
+
+      context('with no allocation', function() {
+        beforeEach(function() {
+          res = {
+            locals: {
+              move: {
+                id: '12345',
+              },
+            },
+            redirect: sinon.stub(),
+          }
+          controller.checkAllocation({}, res, next)
+        })
+
+        it('should redirect to allocation', function() {
+          expect(res.redirect).to.be.calledOnceWithExactly('/move/12345')
+        })
+
+        it('should not call next', function() {
+          expect(next).to.not.be.called
+        })
+      })
+
+      context('with allocation but not person', function() {
+        beforeEach(function() {
+          res = {
+            locals: {
+              move: {
+                id: '12345',
+                allocation: { id: '6789' },
+              },
+            },
+            redirect: sinon.stub(),
+          }
+          controller.checkAllocation({}, res, next)
+        })
+
+        it('should redirect to allocation', function() {
+          expect(res.redirect).to.be.calledOnceWithExactly('/allocation/6789')
+        })
+
+        it('should not call next', function() {
+          expect(next).to.not.be.called
+        })
+      })
+
+      context('with allocation and person', async function() {
+        beforeEach(function() {
+          res = {
+            locals: {
+              move: {
+                id: '12345',
+                allocation: { id: '6789' },
+                person: { id: '__person__' },
+              },
+            },
+            redirect: sinon.stub(),
+          }
+
+          controller.checkAllocation({}, res, next)
+        })
+
+        it('should not redirect', function() {
+          expect(res.redirect).to.not.be.called
+        })
+
+        it('should call next', function() {
+          expect(next).to.be.calledOnceWithExactly()
+        })
+      })
+    })
+
+    describe('#setMoveRelationships()', function() {
```

Comment:
This method is missing the failure test if something goes wrong with the promise.

However, you may be able to avoid this altogether if you don't need to make this extra API call - that would be the much preferred solution in my opinion.

Reply:
👍 to the failure test